File: lib/src/application/commands/foods.dart
import 'package:functionally/reader_stream.dart';

import '../../domain/food.dart';

final class FoodData {
  FoodData({required this.name});

  final String name;
}

Food toFoodEntity(
  FoodData foodData,
) =>
    const Food('');

typedef Foods<FOODS> = ReaderStream<FOODS, Iterable<FoodData>>;

typedef FoodsWithDeps = Stream<Iterable<FoodData>>;

File: src/application/commands/foods.ts
import { Reader } from 'fp-ts/Reader'
import { Observable } from 'rxjs'

import { Food } from '@/domain/food'

export interface FoodData {
	readonly name: string
}

export type Foods = Observable<FoodData[]>
export type FoodsWithDeps<ENV> = Reader<ENV, Observable<FoodData[]>>

export function toFoodEntity(foodDatas: FoodData[]): Food[] {
	return [{ name: 'asd' }]
}
